pkg/metrics/metrics.go

```diff
@@ -259,16 +275,27 @@ func (c *client) WriteMetric(ctx context.Context, name string, count int64) erro
 //	    // handle error
 //	  }
 //	}()
-func (c *client) WriteMetricAsync(ctx context.Context, name string, count int64) func() error {
-	errCh := make(chan error, 1)
+func (c *client) WriteMetricAsync(ctx context.Context, name string, count int64) {
+	if c.OptOut {
+		return
+	}
+	c.AsyncRunners.Add(1)
 	go func() {
-		defer close(errCh)
-		errCh <- c.WriteMetric(ctx, name, count)
+		if err := c.WriteMetric(ctx, name, count); err != nil && c.ErrorHandler != nil {
+			c.ErrorHandler(ctx, err)
+		}
+		c.AsyncRunners.Done()
```

Comment:
Maybe `defer` this call to Done() so the waitgroup will unblock even if if the error handler panics.

Reply:
Good point